xml_converter/src/packaging_protobin.cpp

```diff
@@ -284,3 +291,47 @@ void write_protobuf_file_per_map_id(
             &state);
     }
 }
+void _category_filter_by_depth(
+    const map<string, Category>* marker_categories,
+    int target_depth,
+    map<string, StringHierarchy>* category_hierarchy,
+    vector<string> parent_name,
+    int current_depth = 0) {
+    for (auto it = marker_categories->begin(); it != marker_categories->end(); it++) {
+        vector<string> child_name = parent_name;
+        child_name.push_back(it->first);
+        if (target_depth == current_depth) {
+            StringHierarchy category_filter;
+            category_filter.add_path(child_name, true);
+            (*category_hierarchy)[join(child_name, "_")] = category_filter;
+        }
+
+        _category_filter_by_depth(&(it->second.children), target_depth, category_hierarchy, child_name, current_depth + 1);
+    }
```

Comment:
You have no terminal case here to exit the loop when the correct depth is hit, meaning that this will always recurse over the entire node tree regardless of the depth, but only act on the single depth `target_depth`. The `if` should probably contain a `continue` or the recursion should be in an else, or something similar.

Reply:
I will add continue. No need to check the children if the depth is reached.